### ISSUE TITLE
template_dir schema not found in memory error

### ISSUE THREAD

**Issue by behzade** [ariga/atlas#2112]
Hi. I've been trying to make use of template_dir for applying schema but there seems to be an issue in loading the tempalte dir to memory.
atlas.hcl
```hcl
variable "schema" {
    type = string
    default = "default_db"
}

variable "path" {
    type = string
    description = "template dir path"
    default = "schema.sql"
}

data "template_dir" "schema" {
    path = var.path
    vars = {
        schema = var.schema
    }
}

env "dev" {
    src = data.template_dir.schema.url
    schemas = [var.schema]
    url = "mysql://root:root@mysql:3306"
    dev = "mysql://root:root@mysql_dev:3306"
}
```
schema.sql
```sql
create database {{ .schema }};

create table {{ .schema }}.user (
    `id` bigint unsigned not null auto_increment,
    `name` varchar(255) not null,
    `created_at` datetime NOT NULL DEFAULT CURRENT_TIMESTAMP,
    `updated_at` datetime NOT NULL DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP,
    PRIMARY KEY(`id`)
);
```

I'm getting the following error
```
 data-source state "mem://schema.sql/schema" not found in memory
```
I can't seem to figure out the problem in the config. I would appreciate any help

**Comment by a8m** [comment]
Hey, @behzade 👋 

The `template_dir` is primarily used to compute migration directories. Are you intending to use it as a schema or desired state?

**Comment by behzade** [comment]
Yeah I'm trying to use it to represent the desired state. There is a section in the docs that showcases such a use case.
https://atlasgo.io/atlas-schema/sql#template-directory

**Comment by a8m** [comment]
Got it, thanks. Let me see if I can send a patch for this really quickly

**Comment by behzade** [comment]
Thanks.

**Comment by a8m** [comment]
The issue was resolved with #2113. A new version will be released soon. 

Thanks for reporting this, @behzade.
